### ISSUE TITLE
bug: DAP debug configuration didn't automatically effective

### ISSUE THREAD

**Issue by buptlibo** [nvim-java/nvim-java#87]
### Did you check docs and existing issues?

- [X] I have read all the plugin docs
- [X] I have searched the existing issues
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.9.4

### Operating system/version

MacOS 14.2

### Describe the bug

When I wanto start the springboot project, DAP shows that:
No configuration found for `java`. You need to add configs to `dap.configurations.java` (See `:h dap-configuration`)

My neovim dots is here: https://github.com/buptlibo/nvimdots

### Steps To Reproduce

Use neovim to open a springboot project, then start debug the application.

### Expected Behavior

The project can start normally.

**Comment by s1n7ax** [comment]
@buptlibo Hi, You can find the logs in the .local/share/nvim/nvim-java.log. Can you remove the file, then open up a java project and send the logs

**Comment by buptlibo** [comment]
> @buptlibo Hi, You can find the logs in the .local/share/nvim/nvim-java.log. Can you remove the file, then open up a java project and send the logs

Here is the log
[nvim-java.log](https://github.com/nvim-java/nvim-java/files/13692250/nvim-java.log)

**Comment by s1n7ax** [comment]
@buptlibo 
Did you open a .java file? If so it seems that you have not run `require('lspconfig').jdtls.setup({})`. There is no indication of running that. You can just paste the content of the log btw.

**Comment by buptlibo** [comment]
> @buptlibo Did you open a .java file? If so it seems that you have not run `require('lspconfig').jdtls.setup({})`. There is no indication of running that. You can just paste the content of the log btw.

Yes, I opened the SpringbootApplication.java. 
I add the nvim-java plugin like the wiki "https://github.com/nvim-java/nvim-java/wiki/Lazyvim".

Just now, I add the 
"require("java").setup()
require("lspconfig").jdtls.setup({})" to my config, and the log looks like
`[INFO  六 12/16 18:47:34 2023] ...e/nvim/site/lazy/nvim-java/lua/java/startup/nvim-dep.lua:24: check neovim plugin dependencies
[INFO  六 12/16 18:47:34 2023] ...lazy/nvim-java/lua/java/startup/lspconfig-setup-wrap.lua:11: wrap lspconfig.java.setup function to inject a custom java config
[INFO  六 12/16 18:47:34 2023] ...ocal/share/nvim/site/lazy/nvim-java/lua/java/api/dap.lua:13: add LspAttach event handlers to setup dap adapter & config
[INFO  六 12/16 18:47:34 2023] ...e/nvim/site/lazy/nvim-java/lua/java/startup/nvim-dep.lua:24: check neovim plugin dependencies
[INFO  六 12/16 18:47:34 2023] ...lazy/nvim-java/lua/java/startup/lspconfig-setup-wrap.lua:11: wrap lspconfig.java.setup function to inject a custom java config
[INFO  六 12/16 18:47:34 2023] ...ocal/share/nvim/site/lazy/nvim-java/lua/java/api/dap.lua:13: add LspAttach event handlers to setup dap adapter & config
[INFO  六 12/16 18:47:34 2023] ...e/nvim/site/lazy/nvim-java/lua/java/startup/nvim-dep.lua:24: check neovim plugin dependencies
[INFO  六 12/16 18:47:34 2023] ...lazy/nvim-java/lua/java/startup/lspconfig-setup-wrap.lua:11: wrap lspconfig.java.setup function to inject a custom java config
[INFO  六 12/16 18:47:34 2023] ...ocal/share/nvim/site/lazy/nvim-java/lua/java/api/dap.lua:13: add LspAttach event handlers to setup dap adapter & config
`

**Comment by s1n7ax** [comment]
Still there is no any indication of you calling `lspconfig.jdtls.setup({})`. This is something you should do after setting up nvim-java. Note that lazyvim configuration might not work for you because every one setup things differently. You can try calling `lspconfig.jdtls.setup({})` after `java.setup()`.

Your dotfile is too big. If you can provide me a config without a lot of plugins and stuff I might be able to help you. BTW, I tried cloning and it fails at some point. Probably `.git` is broken.

**Comment by buptlibo** [comment]
Maybe I found the problem in my nvim dotfiles. I use the lazy.nvim, if I set the nvim-java lazy=true(even if I add the ft="java"), the dap configuration will not automcatically work, but when I remove the the lazy setting, it works(sometime, if set lazy=false, the neovim start throw error of nvim-dap. When I enter quickly, the dap config will not auto config, but if I wait a little while, then enter, the java dap configured success). 

If I solved this problem, I will reply here, it may take a lot of time, cause I'm no familiar with neovim and the lua configuration. My nvim dotfiles is forked from others.
